targets/targets.json

```diff
@@ -8353,8 +8353,7 @@
         "macros_add": ["CMSDK_CM7"],
         "device_has_add": ["MPU"]
     },
-    "NUMAKER_PFM_M2351": {
-        "core": "Cortex-M23-NS",
+    "NUMAKER_PFM_M2351_CM": {
```

Comment:
This change needs to be replicated into the online database (please ask you TAM representative) and the platform database in https://github.com/ARMmbed/mbed-os-tools

Reply:
> This change needs to be replicated into the online database (please ask you TAM representative)

OK

> the platform database in https://github.com/ARMmbed/mbed-os-tools

The platform name **NUMAKER_PFM_M2351** in mbed-os-tools database is kept unchanged for compatibility. Don't want to meet mismatch issue with mbed-os-tools and mbed-os.

Reply:
I'm not sure what your comments mean, can you explain? The names need the same in all places or the entire build system, testing and reporting will not work.

Reply:
@madchutney Originally, I want to keep the [M2351 platform name in mbed-os-tools](https://github.com/ARMmbed/mbed-os-tools/blob/e732afb07a8bfc3796a522074c4ae4ad27b19637/src/mbed_os_tools/detect/platform_database.py#L227) unchanged, so users don't need to care about version match between mbed-os and mbed-os-tools. Now that the name must be consistent across different tools, I will send another PR to mbed-os-tools repo to change platform name to **NU_PFM_M2351** from **NUMAKER_PFM_M2351** after this PR is confirmed. https://github.com/ARMmbed/mbed-os/pull/11288/commits/bcf8cc63b581d03be0d54ec4410934abdccb16af is addressing the target name/platform name conversion for M2351 like other PSA targets.